lib/manager/npm/extract/yarn.ts

```diff
@@ -36,3 +41,26 @@ export async function getYarnLock(filePath: string): Promise<LockFile> {
     return { isYarn1: true, lockedVersions: {} };
   }
 }
+
+export function getZeroInstallPaths(yarnrcYml: string): string[] {
+  const conf = parseSyml(yarnrcYml);
+  const paths = [conf.cacheFolder || './.yarn/cache', '.pnp.cjs', '.pnp.js'];
+  if (miscUtils.tryParseOptionalBoolean(conf.pnpEnableInlining) === false) {
+    paths.push(conf.pnpDataPath || './.pnp.data.json');
+  }
+  return paths;
+}
+
+export async function isZeroInstall(yarnrcYmlPath: string): Promise<boolean> {
+  const yarnrcYml = await readLocalFile(yarnrcYmlPath, 'utf8');
+  if (is.string(yarnrcYml)) {
```

Comment:
May use `is.nonEmptyString` ?

Reply:
@viceice An empty YAML is still valid and in this case, it should check for default paths.